File: src/dataset/typescript/pangram_exercism-javascript-samajammin/claude/ClaudeSonnet4_pangram_v1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class Pangram {
    constructor(phrase) {
        this.phrase = phrase.toLowerCase();
    }
    isPangram() {
        let letterMask = 0;
        for (let i = 0; i < this.phrase.length; i++) {
            const charCode = this.phrase.charCodeAt(i);
            if (charCode >= 97 && charCode <= 122) { // a-z
                letterMask |= 1 << (charCode - 97);
                if (letterMask === 0x3ffffff)
                    return true; // All 26 bits set
            }
        }
        return false;
    }
}
exports.default = Pangram;

File: src/dataset/c/pangram_exercism-Manoloon/claude/ClaudeSonnet4_pangram_v1.c
#include "pangram.h"
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

bool is_pangram(const char *sentence) 
{ 
    if(sentence == NULL) return false;
    
    unsigned int marked = 0;
    const unsigned int all_letters = 0x3FFFFFF; // 26 bits set
    
    for(const char *p = sentence; *p && marked != all_letters; ++p)
    {
        if(*p >= 'A' && *p <= 'Z')
        {
            marked |= 1U << (*p - 'A');
        }
        else if(*p >= 'a' && *p <= 'z')
        {
            marked |= 1U << (*p - 'a');
        }
    }
    
    return marked == all_letters;
}